service/telemetry.go

```diff
@@ -109,8 +110,22 @@ func (tel *colTelemetry) initOnce(col *Collector) error {
 
 	logger.Info("Setting up own telemetry...")
 
-	instanceUUID, _ := uuid.NewRandom()
-	instanceID := instanceUUID.String()
+	if resource == nil {
+		resource = map[string]string{}
+	}
+
+	var instanceID string
+	instanceID = resource[semconv.AttributeServiceInstanceID]
+	if instanceID == "" {
+		// AttributeServiceInstanceID is not specified in the config. Auto-generate one.
+		instanceUUID, _ := uuid.NewRandom()
+		instanceID = instanceUUID.String()
+		resource[semconv.AttributeServiceInstanceID] = instanceID
+	}
+
+	if AddCollectorVersionTag {
```

Comment:
I think we need an issue to make this feature gate or remove it, since I believe nobody sets this.

Reply:
Yes, that's why I asked for an "issue" :)

Reply:
Issue created: https://github.com/open-telemetry/opentelemetry-collector/issues/5404